desc/compute/_geometry.py

```diff
@@ -0,0 +1,354 @@
+    label="R_{0} / a",
+    units="~",
+    units_long="None",
+    description="Aspect ratio",
+    dim=0,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="",
+    data=["R0", "a"],
+)
+def _R0_over_a(params, transforms, profiles, data, **kwargs):
+    data["R0/a"] = data["R0"] / data["a"]
+    return data
+
+
+@register_compute_fun(
+    name="n_rho",
+    label="n_{rho}",
+    units="~",
+    units_long="None",
+    description="Unit normal vector to constant rho surface",
+    dim=3,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["e^rho"],
+)
+def _n_rho(params, transforms, profiles, data, **kwargs):
+    data["n_rho"] = (data["e^rho"].T / jnp.linalg.norm(data["e^rho"])).T
+    return data
+
+
+@register_compute_fun(
+    name="L_sff",
+    label="L_{sff}",
+    units="m",
+    units_long="meters",
+    description="L coefficient of second fundamental form",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["n_rho", "e_theta_t"],
+)
+def _L_sff(params, transforms, profiles, data, **kwargs):
+    data["L_sff"] = dot(data["e_theta_t"], data["n_rho"])
+    return data
+
+
+@register_compute_fun(
+    name="M_sff",
+    label="M_{sff}",
+    units="m",
+    units_long="meters",
+    description="M coefficient of second fundamental form",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["n_rho", "e_theta_z"],
+)
+def _M_sff(params, transforms, profiles, data, **kwargs):
+    data["M_sff"] = dot(data["e_theta_z"], data["n_rho"])
+    return data
+
+
+@register_compute_fun(
+    name="N_sff",
+    label="N_{sff}",
+    units="m",
+    units_long="meters",
+    description="N coefficient of second fundamental form",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["n_rho", "e_zeta_z"],
+)
+def _N_sff(params, transforms, profiles, data, **kwargs):
+    data["N_sff"] = dot(data["e_zeta_z"], data["n_rho"])
+    return data
+
+
+@register_compute_fun(
+    name="curvature_k1",
+    label="k_{1}",
+    units="m^{-1}",
+    units_long="Inverse meters",
+    description="First principle curvature of flux surfaces",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["g_tt", "g_tz", "g_zz", "L_sff", "M_sff", "N_sff"],
+)
+def _curvature_k1(params, transforms, profiles, data, **kwargs):
+    # following notation from
+    # https://en.wikipedia.org/wiki/Parametric_surface#Curvature
+    E = data["g_tt"]
+    F = data["g_tz"]
+    G = data["g_zz"]
+    L = data["L_sff"]
+    M = data["M_sff"]
+    N = data["N_sff"]
+    a = E * G - F**2
+    b = F * M - L * G - E * N
+    c = L * N - M**2
+    r1 = (-b + jnp.sqrt(b**2 - 4 * a * c)) / (2 * a)
+    r2 = (-b - jnp.sqrt(b**2 - 4 * a * c)) / (2 * a)
+    data["curvature_k1"] = jnp.maximum(r1, r2)
+    data["curvature_k2"] = jnp.minimum(r1, r2)
```

Comment:
because the calculation of k1 and k2 are nearly identical (and I didn't really want to add more intermediate quantities to the data index), I thought it would be best to always return both from either function, so that in theory only one of them ever needs to be called.